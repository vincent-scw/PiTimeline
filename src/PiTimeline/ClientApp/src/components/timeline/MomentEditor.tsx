import React, { useState } from "react";
import { RichTextInput, TDatePicker } from "../controls";
import { toast } from "react-toastify";
import * as Svc from '../../services';
import Popup from "reactjs-popup";
import { TGallery } from "../gallery/TGallery";

export interface MomentEditorProps {
  moment?: Svc.Moment;
  saved?: Function;
}

export const MomentEditor: React.FC<MomentEditorProps> = (props) => {
  const [moment, setMoment] = useState<Svc.Moment>(props.moment || {});
  const [popupOpen, setPopupOpen] = useState<boolean>(false);

  const stateChanged = (prop: string, v: any) => {
    let newEntity = { ...moment, [prop]: v };
    setMoment(newEntity);
  }

  const saveMoment = () => {
    if (moment.id) {
      Svc.MomentSvc.updateMoment(moment)
        .then(t => {
          toast.info(`${t.data.name} has been updated.`)
          if (props.saved) props.saved(t.data);
        });
    } else {
      Svc.MomentSvc.createMoment(moment)
        .then(t => {
          toast.info(`${t.data.name} has been created.`)
          if (props.saved) props.saved(t.data);
        });
    }
  }

  const insertImage = (): string => {
    setPopupOpen(true)
    return '';
  }

  const insertVideo = (): string => {
    setPopupOpen(true)
    return '';
  }

  return (
    <React.Fragment>
      <section>
        <p className="subtitle">
          Moment Editor
        </p>
      </section>
      <hr />
      <form>
        <TDatePicker name="Date"
          value={moment.takePlaceAtDateTime}
          valueChanged={(date) => stateChanged('takePlaceAtDateTime', date)}></TDatePicker>
        <RichTextInput value={moment.content} valueChanged={(c) => stateChanged('content', c)} 
          insertImageClicked={() => insertImage()}
          insertVideoClicked={() => insertVideo()} />
        <div className="field">
          <div className="control">
            <a className="button is-primary is-small is-fullwidth"
              onClick={saveMoment}>
              Save
            </a>
          </div>
        </div>
      </form>
      <Popup open={popupOpen} closeOnDocumentClick onClose={() => setPopupOpen(false)}>
        <TGallery />
      </Popup>
    </React.Fragment>
  );
}