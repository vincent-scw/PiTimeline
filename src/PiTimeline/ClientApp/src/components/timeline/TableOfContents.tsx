import React, { useEffect, useRef, useState } from "react";
import Moment from 'react-moment';

export interface TableOfContentsProps {
  headers: any[];
}

export const TableOfContents: React.FC<TableOfContentsProps> = (props) => {
  const { headers } = props;
  const [activeId, setActiveId] = useState();

  const headerElementsRef = useRef({});

  useEffect(() => {
    if (!headers || headers.length === 0) return;

    const callback = (headings) => {
      headerElementsRef.current = headings.reduce((map, headerElement) => {
        map[headerElement.target.id] = headerElement;
        return map;
      }, headerElementsRef.current);

      const visibleHeaders = [];
      Object.keys(headerElementsRef.current).forEach((key) => {
        const headerElement = headerElementsRef.current[key];
        if (headerElement.isIntersecting)
          visibleHeaders.push(headerElement);
      });
      const getIndexFromId = (id) =>
        headerElements.findIndex((header) => header.id === id);

      if (visibleHeaders.length === 1) {
        setActiveId(visibleHeaders[0].target.id);
      } else if (visibleHeaders.length > 1) {
        const sortedVisibleHeaders = visibleHeaders.sort(
          (a, b) => getIndexFromId(a.target.id) - getIndexFromId(b.target.id)
        );
        setActiveId(sortedVisibleHeaders[0].target.id);
      }
    }

    const observer = new IntersectionObserver(callback, {
      rootMargin: '-80px 0px -40% 0px',
    });

    const headerElements = Array.from(document.querySelectorAll(".scrollable-header"));
    headerElements.forEach((element) => observer.observe(element));

    return () => { headerElementsRef.current = {}; observer.disconnect(); }
  }, [headers, setActiveId]);

  const headClicked = (e, id: string) => {
    e.preventDefault();
    document.querySelector(`#${id}`).scrollIntoView({ behavior: 'smooth' });
  }

  const buildHeader = () => (
    <ul>
      {headers.map(h => {
        const headerId = `gm-${h.id}`;
        return (
          <li key={h.id} className={`${headerId === activeId ? "is-current" : "is-normal"} level-1`}>
            <div>
              <div></div>
              <a href={`#${h.group}`} onClick={e => headClicked(e, headerId)}>{h.group}</a>
            </div>
            {h.moments.length > 0 && (
              <ul>
                {h.moments.map(m => {
                  const l2HeaderId = `m-${m.id}`;
                  return (
                    <li key={m.id} className={`${l2HeaderId === activeId ? "is-current" : "is-normal"} level-2`}>
                      <div>
                        <div></div>
                        <a onClick={e => headClicked(e, l2HeaderId)}><Moment format="ddd MMM DD">{m.takePlaceAtDateTime}</Moment></a>
                      </div>
                    </li>);
                })}
              </ul>
            )}
          </li>
        )
      })
      }
    </ul>
  );

  return (
    <nav aria-label="Table of contents" className="toc">
      {headers && buildHeader()}
    </nav>
  )
}