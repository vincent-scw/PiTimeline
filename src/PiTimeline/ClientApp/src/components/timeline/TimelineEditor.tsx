import React, { useState } from "react";
import { useDispatch } from "react-redux";
import { TextInput } from "../controls";
import * as Svc from '../../services';
import { createTimeline, updateTimeline } from "../../services";

export interface TimelineEditorProps {
  timeline?: Svc.Timeline;
  saved?: Function;
}

export const TimelineEditor: React.FC<TimelineEditorProps> = (props) => {
  const dispatch = useDispatch();
  const [timeline, setTimeline] = useState<Svc.Timeline>(props.timeline || {});

  const stateChanged = (prop: string, v: any) => {
    let newEntity = { ...timeline, [prop]: v };
    setTimeline(newEntity);
  }

  const saveTimeline = () => {
    if (timeline.id) {
      dispatch(updateTimeline(timeline));
    } else {
      dispatch(createTimeline(timeline));
    }
  }

  return (
    <React.Fragment>
      <form>
        <TextInput
          valueChanged={(v) => stateChanged('title', v)}
          value={timeline.title}
          placeholder="Input timeline title here" />

        <div className="field">
          <div className="control">
            <a className="button is-primary is-small is-fullwidth"
              onClick={saveTimeline}>
              Save
            </a>
          </div>
        </div>
      </form>
    </React.Fragment>
  )
}