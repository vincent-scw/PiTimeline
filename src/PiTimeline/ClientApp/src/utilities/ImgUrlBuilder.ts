export const buildImgUrl = (path: string, resFactor?: number) => {
  let url = `api/Gallery/f/${path}`;
  if (resFactor)
    url += `?res=${resFactor}`;

  return url;
}