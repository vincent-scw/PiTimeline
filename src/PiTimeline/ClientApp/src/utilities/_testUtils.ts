import React from 'react'
import { Provider } from 'react-redux'
import { Timeline, Moment } from '../services';

export const createMockTimeline = (): Timeline => {
  return {
    id: 'id',
    title: 'timeline',
    description: 'desc',
    coverPatternUrl: 'http://mock',
    since: new Date('2022-06-01T00:00:00.000Z') // 01 Jun 2022
  };
};

export const createMockMoment = (): Moment => {
  return {
    id: 'mid',
    content: 'content',
    takePlaceAtDateTime: new Date('2022-06-01T00:00:00.000Z'), // 01 Jun 2022,
    timelineId: 'id'
  };
}