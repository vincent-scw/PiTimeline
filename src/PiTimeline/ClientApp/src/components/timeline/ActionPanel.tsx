import React from "react";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { Link } from "react-router-dom";
import { faPlus, faImages } from '@fortawesome/free-solid-svg-icons';
import { Popup } from "reactjs-popup";
import { MomentEditor } from './MomentEditor';
import { Moment, Timeline } from "../../services";

export interface ActionPanelProps {
  timeline: Timeline;
}

export const ActionPanel: React.FC<ActionPanelProps> = (props) => {
  const { timeline } = props;
  const newMoment: Moment = { timelineId: timeline.id, content: '', takePlaceAtDateTime: new Date() };

  return (
    <div className="panel left-dockbg">
      <p className="panel-block dock-img-block">
        <Popup position="center center" modal={true} closeOnDocumentClick={false}
          trigger={<a>
            <span className="icon has-text-success">
              <FontAwesomeIcon icon={faPlus} />
            </span>
          </a>}>
          {close => <MomentEditor saved={(t) => { close(); }} moment={newMoment} />
          }
        </Popup>
      </p>
      <p className="panel-block dock-img-block">
        <Link to={'/g'}>
          <FontAwesomeIcon icon={faImages} />
        </Link>
      </p>
    </div>
  );
}