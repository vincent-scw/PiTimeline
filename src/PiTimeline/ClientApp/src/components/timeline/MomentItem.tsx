import React from "react";
import Moment from 'react-moment';
import Popup from "reactjs-popup";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faEdit, faTrashAlt } from "@fortawesome/free-solid-svg-icons";
import { MomentEditor } from "./MomentEditor";
import * as Svc from "../../services";

export interface MomentItemProps {
  moment: Svc.Moment;
  deleteMoment?: Function
}

export const MomentItem: React.FC<MomentItemProps> = (props) => {
  const { moment, deleteMoment } = props;

  return (
    <div className="timeline-item">
      <div className="timeline-marker"></div>
      <div className="timeline-content">
        <p className="heading scrollable-header" id={`m-${moment.id}`}>
          <Moment format="ddd MMM DD">{moment.takePlaceAtDateTime}</Moment>
        </p>
        <p dangerouslySetInnerHTML={{ __html: moment.content }} />
        <div className="level is-mobile">
          <div className="level-left"></div>
          <div className="level-right">
            <Popup position="center center" modal={true} closeOnDocumentClick={false}
              trigger={
                <a className="level-item" >
                  <span className="icon has-text-info">
                    <FontAwesomeIcon icon={faEdit} />
                  </span>
                </a>
              }>
              {close => <MomentEditor done={close} moment={moment} />}
            </Popup>

            <a onClick={() => deleteMoment(moment)}>
              <span className="icon has-text-grey-light">
                <FontAwesomeIcon icon={faTrashAlt} />
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  )
}