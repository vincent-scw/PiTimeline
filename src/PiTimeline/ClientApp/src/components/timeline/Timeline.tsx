import React, { useEffect } from "react";
import { useParams } from "react-router-dom";
import Moment from 'react-moment';
import { ActionPanel } from './ActionPanel';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faEdit, faTrashAlt } from "@fortawesome/free-solid-svg-icons";
import Popup from "reactjs-popup";
import { MomentEditor } from "./MomentEditor";
import { useSelector, useDispatch } from "react-redux";
import { 
  getTimelineDetail, 
  deleteMoment,
  selectGroupedMoments, 
  selectTimeline, 
  selectTimelineIsLoading } from "../../services";

const Timeline: React.FC = () => {
  const { tid } = useParams<any>();

  const dispatch = useDispatch();
  const timeline = useSelector(selectTimeline);
  const isLoading = useSelector(selectTimelineIsLoading);
  const groupedMoments = useSelector(selectGroupedMoments);

  useEffect(() => {
    dispatch(getTimelineDetail(tid));
  }, [tid, dispatch])

  const delMoment = (moment) => {
    dispatch(deleteMoment(moment));
  }

  return (
    <div>
      {isLoading && <div>Loading...</div>}
      {timeline &&
        <div>
          <div className="t-content">
            <section>
              <h1 className="title">
                {timeline.title}
              </h1>
            </section>
          </div>
          {groupedMoments.map(gm => (
            <div className="timeline" key={gm.group}>
              <header className="timeline-header">
                <span className="tag is-primary">{gm.group}</span>
              </header>
              {gm.moments.map(m => (
                <div className="timeline-item" key={m.id}>
                  <div className="timeline-marker"></div>
                  <div className="timeline-content">
                    <p className="heading">
                      <Moment format="ddd MMM DD">{m.takePlaceAtDateTime}</Moment>
                    </p>
                    <p dangerouslySetInnerHTML={{ __html: m.content }} />
                    <div className="level is-mobile">
                      <div className="level-left"></div>
                      <div className="level-right">
                        <Popup position="center center" modal={true} closeOnDocumentClick={false}
                          trigger={
                            <a className="level-item" >
                              <span className="icon has-text-info">
                                <FontAwesomeIcon icon={faEdit} />
                              </span>
                            </a>
                          }>
                          {close => <MomentEditor done={close} moment={m} />}
                        </Popup>

                        <a onClick={() => delMoment(m)}>
                          <span className="icon has-text-grey-light">
                            <FontAwesomeIcon icon={faTrashAlt} />
                          </span>
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              ))}
            </div>
          ))
          }
          <div className="timeline">
            <div className="timeline-header">
              <span className="tag is-medium is-primary"><img src="../assets/favicon.png"></img></span>
            </div>
          </div>
          <ActionPanel timeline={timeline} />
        </div>
      }
    </div>
  );
}

export default Timeline;