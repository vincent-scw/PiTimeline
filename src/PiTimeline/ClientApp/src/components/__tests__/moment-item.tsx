import React from 'react';
import { mount } from 'enzyme';
import { MomentItem } from '../timeline/MomentItem';
import { createMockMoment } from '../../utilities/_testUtils';

describe('moment-item', () => {
  it('renders correctly', () => {
    const moment = createMockMoment();
    const wrapper = mount(<MomentItem moment={moment}/>);

    expect(wrapper).toMatchSnapshot();
  })
})