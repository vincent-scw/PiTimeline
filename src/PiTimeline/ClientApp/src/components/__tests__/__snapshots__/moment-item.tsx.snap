// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`moment-item renders correctly 1`] = `
<MomentItem
  moment={
    Object {
      "content": "content",
      "id": "mid",
      "takePlaceAtDateTime": 2022-06-01T00:00:00.000Z,
      "timelineId": "id",
    }
  }
>
  <div
    className="timeline-item"
  >
    <div
      className="timeline-marker"
    />
    <div
      className="timeline-content"
    >
      <p
        className="heading scrollable-header"
        id="m-mid"
      >
        <n
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="ddd MMM DD"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          trim={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <time
            dateTime={"2022-06-01T00:00:00.000Z"}
          >
            Wed Jun 01
          </time>
        </n>
      </p>
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": "content",
          }
        }
      />
      <div
        className="level is-mobile"
      >
        <div
          className="level-left"
        />
        <div
          className="level-right"
        >
          <ForwardRef
            closeOnDocumentClick={false}
            modal={true}
            position="center center"
            trigger={
              <a
                className="level-item"
              >
                <span
                  className="icon has-text-info"
                >
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          576,
                          512,
                          Array [],
                          "f044",
                          "M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z",
                        ],
                        "iconName": "edit",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    swapOpacity={false}
                    symbol={false}
                    title=""
                    transform={null}
                  />
                </span>
              </a>
            }
          >
            <a
              aria-describedby="popup-1"
              className="level-item"
              key="T"
              onClick={[Function]}
            >
              <span
                className="icon has-text-info"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        576,
                        512,
                        Array [],
                        "f044",
                        "M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z",
                      ],
                      "iconName": "edit",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-edit fa-w-18 "
                    data-icon="edit"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 576 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
              </span>
            </a>
          </ForwardRef>
          <a
            onClick={[Function]}
          >
            <span
              className="icon has-text-grey-light"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f2ed",
                      "M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z",
                    ],
                    "iconName": "trash-alt",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                swapOpacity={false}
                symbol={false}
                title=""
                transform={null}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-trash-alt fa-w-14 "
                  data-icon="trash-alt"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </FontAwesomeIcon>
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</MomentItem>
`;
